### PR TITLE
containers for assembly pipeline

### PR DESCRIPTION
**Pull request by Vkale1**

### CODE REVIEW

File: assembly_pipeline_bwa/map_host.sh [hunk 1]
```diff
@@ -0,0 +1,72 @@
+#!/bin/bash
+
```
Comment (mberacochea) [inline assembly_pipeline_bwa/map_host.sh:2]: Add `set -e` to make the script fail if one of the commands in it fails.

File: assembly_pipeline_bwa/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,43 @@
+# Docker container for BWA
+
+FROM ubuntu:20.04
```
Comment (mberacochea) [inline assembly_pipeline_bwa/Dockerfile:3]: This looks _very_ similar to https://github.com/EBI-Metagenomics/containers/blob/master/bwamem2/Dockerfile

Comment (mberacochea) [reply assembly_pipeline_bwa/Dockerfile:3]: Maybe we should merge them

File: assembly_pipeline_python3_scripts/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,16 @@
+FROM python:3.7.9-slim-buster
+
+#tag microbiome-informatics/assembly-pipeline.python:3.7.9
```
Comment (mberacochea) [inline assembly_pipeline_python3_scripts/Dockerfile:3]: It's missing the LICENCE and VERSION

File: assembly_pipeline_bwa/Dockerfile (hunk 2 above)
Comment (Vkale1) [reply assembly_pipeline_bwa/Dockerfile:3]: It is indeed identical! I just needed the script to be added to the container. I can add it to the other one - but we were unsure of tracking back where the scripts come from. Option to also put the scripts in separate directories within the container?

File: assembly_pipeline_readfq/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,17 @@
+FROM alpine:3.16.0
+
+#tag  microbiome-informatics/assembly-pipeline.readfq:latest
```
Comment (mberacochea) [inline assembly_pipeline_readfq/Dockerfile:3]: It's missing some metadata, LICENCE, VERSION, LINK to the tool.
Also, the folder should be `readfq`.

File: assembly_pipeline_readfq/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+COPY kseq.h \
+     kseq_fastq_base.c \
+/tools/
+
+RUN chmod a+x /tools/*
+
+RUN gcc -o kseq_fastq_base /tools/kseq_fastq_base.c -lz
+
+#CMD ["./kseq_fastq_base"]
```
Comment (mberacochea) [inline assembly_pipeline_readfq/Dockerfile:17]: Please, remove the commented code.

**Comment by mberacochea** [comment]
The docker files look good, BWA seems to be duplicated and I've added some minor comments.

We should try to have the least amount of containers possible. For example one BWA that is used in several pipelines, and not different BWA containers per pipeline.

File: assembly_pipeline_readfq/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,17 @@
+FROM alpine:3.16.0
```
[the remark this answers is not in the document]
Comment (Vkale1) [reply assembly_pipeline_readfq/Dockerfile]: readfq removed

File: assembly_pipeline_python3_scripts/Dockerfile [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+LABEL software="assembly-pipeline.python"
+LABEL software.version="1.0.0"
+LABEL website="https://github.com/EBI-Metagenomics/CWL-assembly/tree/develop/cwl"
+LABEL license="https://github.com/EBI-Metagenomics/CWL-assembly/blob/develop/LICENSE"
+
+# Dependencies
+RUN conda config --add channels bioconda && \
+    conda config --add channels conda-forge
+
+RUN mamba install -c conda-forge -y biopython && \
```
Comment (mberacochea) [inline assembly_pipeline_python3_scripts/Dockerfile:14]: Can you please pin down the version of biopython?

File: bwamem2/Dockerfile [hunk 8]
```diff
@@ -2,6 +2,7 @@
 
 FROM ubuntu:20.04
 
+#tag microbiome-informatics/assembly-pipeline.bwamem2:2.2.1
```
Comment (mberacochea) [inline bwamem2/Dockerfile:5]: Please remove this line :)

File: bwamem2/map_host.sh [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+fi
+
+if [ ${THREADS} -eq 1 ]
+then
+    THREADS_SAM=1
+else
+    THREADS_SAM=$((${THREADS}-1))
+fi
+
+name=${FASTQ_R1%_*}
```
Comment (mberacochea) [inline bwamem2/map_host.sh:52]: There was a bug on this line right?. The basename issue if I remember correctly

**Comment by mberacochea** [comment]
We don't need this container anymore :)

**Pull request closed without merging**